static/app/components/events/eventTags/eventTagsTree.tsx

```diff
@@ -57,136 +86,163 @@ function addToTagTree(
-  content: TagTreeContent;
-  projectId: string;
-  projectSlug: string;
-  streamPath: string;
-  tagKey: string;
-  isLast?: boolean;
-  spacerCount?: number;
-}
-
 function TagTreeRow({
   content,
   tagKey,
   spacerCount = 0,
   isLast = false,
   ...props
 }: TagTreeRowProps) {
-  const subtreeTags = Object.keys(content.subtree);
   const organization = useOrganization();
   const location = useLocation();
   const originalTag = content.originalTag;
 
   return (
-    <Fragment>
-      <TreeRow>
-        <TreeKeyTrunk spacerCount={spacerCount}>
-          {spacerCount > 0 && (
-            <Fragment>
-              <TreeSpacer spacerCount={spacerCount} isLast={isLast} />
-              <TreeBranchIcon />
-            </Fragment>
+    <TreeRow>
+      <TreeKeyTrunk spacerCount={spacerCount}>
+        {spacerCount > 0 && (
+          <Fragment>
+            <TreeSpacer spacerCount={spacerCount} isLast={isLast} />
+            <TreeBranchIcon />
+          </Fragment>
+        )}
+        <TreeKey>{tagKey}</TreeKey>
+      </TreeKeyTrunk>
+      <TreeValueTrunk>
+        <TreeValue>
+          {originalTag ? (
+            <EventTagsContent
+              tag={originalTag}
+              organization={organization}
+              query={generateQueryWithTag(
+                {...location.query, referrer: 'event-tags-tree'},
+                originalTag
+              )}
+              meta={content?.meta ?? {}}
+              {...props}
+            />
+          ) : (
+            content.value
           )}
-          <TreeKey>{tagKey}</TreeKey>
-        </TreeKeyTrunk>
-        <TreeValueTrunk>
-          <TreeValue>
-            {originalTag ? (
-              <EventTagsContent
-                tag={originalTag}
-                organization={organization}
-                query={generateQueryWithTag(
-                  {...location.query, referrer: 'event-tags-tree'},
-                  originalTag
-                )}
-                meta={content?.meta ?? {}}
-                {...props}
-              />
-            ) : (
-              content.value
-            )}
-          </TreeValue>
-        </TreeValueTrunk>
-      </TreeRow>
-      {subtreeTags.map((t, i) => (
-        <TagTreeRow
-          key={`${t}-${i}`}
-          tagKey={t}
-          content={content.subtree[t]}
-          spacerCount={spacerCount + 1}
-          isLast={i === subtreeTags.length - 1}
-          {...props}
-        />
-      ))}
-    </Fragment>
+        </TreeValue>
+      </TreeValueTrunk>
+    </TreeRow>
   );
 }
 
-interface EventTagsTreeProps {
-  projectId: string;
-  projectSlug: string;
-  streamPath: string;
-  tags: EventTag[];
-  meta?: Record<any, any>;
+/**
+ * Function to recursively create a flat list of all rows to be rendered for a given TagTree
+ * @param {TagTreeRowProps} props The props for rendering the root of the TagTree
+ * @returns {React.ReactNode[]} A list of TagTreeRow components to be rendered in this tree
```

Comment:
nit - tsdocs shouldn't redeclare types, just use the typescript types inline
```suggestion
 * @param props The props for rendering the root of the TagTree
 * @getsentry/engineering-stewardship  A list of TagTreeRow components to be rendered in this tree
```